text/000-sort-variable-unification.md

```diff
@@ -0,0 +1,77 @@
+- Title: Introducing quality variables in sort unification
+
+- Drivers: Pierre-Marie Pédrot
+
+----
+
+# Summary
+
+In addition to variables for universe levels, we generalize the type of sorts to
+also contain quality variables, allowing to delay the choice between Prop, Type
+and SProp made by the unification. This CEP is only concerned about unification,
+although the kernel structures need to be modified accordingly, the kernel
+treats undefined quality variables as an error, similarly to evars.
+
+The first part of this CEP is implemented in [Coq PR 16903](https://github.com/coq/coq/pull/16903).
+
+# Motivation
+
+There are several underlying motivations for this CEP.
+
+- Immediate fix: alleviate the longstanding issue of having to explicitly
+annotate variables with `Prop` to make the unification go through.
+- Short term: fix the `SProp` mess in the implementation of kernel and unification.
+- Long term: enhance the modularity of Coq by allowing terms to be polymorphic
+over sorts like `Prop` / `Type` / `SProp`, a.k.a. stop triplicating everything.
```

Comment:
Will this also finally replace template polymorphism, when combined with universe polymorphism?

Reply:
Yes.